### PR TITLE
runtime: Add trusted runtime option

### PR DESCRIPTION
**Pull request by jcvenegas**
Some CRI compatible runtimes may not support provileged operations.
Specifically hypervisor based runtimes (like kata-containers, cc-runtime
and runv) do not support privileged operations like:

Provide access to the host namespaces Create fully privileged containers
with access to host devices.  Hypervisor based runtimes create container
workloads within virtual machines. When running host privileged
containers with them, they wont provide support to requested the
privileged opertations.

This commits add the option to define two runtimes:

trusted runtime : Used when a privileged container is requested.
Default runtime : for non-privileged workloads.

A container that belongs to a privileged pod will inherent this property
an will be created with the trusted runtime.

- Add options to define trusted runtime
- Add logic to decide if a sanbox is trusted
- Export annotation containers below to a trusted sandbox

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

### CODE REVIEW

**Comment by jcvenegas** [comment]
@sameo @egernst  ping please take a look

**Comment by jcvenegas** [comment]
New passing cri-tools using mixed runtimes:
```
CRI_CONTAINERD_FLAGS="--containerd-runtime-engine cc-runtime --containerd-trusted-runtime-engine runc " make test-cri
```
- runtime should ignore a seccomp profile that blocks setting hostname when privileged
- mount with 'rprivate' should not support propagation
 - runtime should support HostPID
- should support propagation from host to container and vice versa
- runtime should support HostIpc is true
- mount with 'rslave' should support propagation from host to container	
 - HostNetwork is true

File: pkg/server/container_create.go [hunk 1]
```diff
@@ -77,6 +77,7 @@ func (c *criContainerdService) CreateContainer(ctx context.Context, r *runtime.C
 	config := r.GetConfig()
 	sandboxConfig := r.GetSandboxConfig()
 	sandbox, err := c.sandboxStore.Get(r.GetPodSandboxId())
+
```
Comment (sameo) [inline pkg/server/container_create.go]: Nit: Unneeded change.

File: cmd/cri-containerd/options/options.go [hunk 2]
```diff
@@ -45,13 +45,21 @@ type ContainerdConfig struct {
 	// Runtime is the runtime to use in containerd. We may support
 	// other runtimes in the future.
 	Runtime string `toml:"runtime" json:"runtime,omitempty"`
+	// TrustedRuntime is the runtime to use for trusted workloads in containerd.
```
Comment (sameo) [inline cmd/cri-containerd/options/options.go]: `with containerd`

File: cmd/cri-containerd/options/options.go [hunk 3]
```diff
@@ -45,13 +45,21 @@ type ContainerdConfig struct {
...
+	TrustedRuntime string `toml:"runtime" json:"runtime,omitempty"`
```
Comment (sameo) [inline cmd/cri-containerd/options/options.go]: Here we should specify that `TrustedRuntime` is optional and when not specified will be either left empty or set to the same value as `Runtime`.

File: cmd/cri-containerd/options/options.go [hunk 4]
```diff
@@ -45,13 +45,21 @@ type ContainerdConfig struct {
...
 	// RuntimeEngine is the name of the runtime engine used by containerd.
 	// Containerd default should be "runc"
 	// We may support other runtime engines in the future.
 	RuntimeEngine string `toml:"runtime_engine" json:"runtimeEngine,omitempty"`
+	// TrustedRuntimeEngine is the name of the runtime engine used by containerd to run trusted workloads.
+	// Containerd default should be "runc"
```
Comment (sameo) [inline cmd/cri-containerd/options/options.go]: If I remember the convoluted containerd code correctly, `RuntimeEngine` will translate in the containerd container runtime path or simply binary name. So I think here we should add more comments to specify that the cri-containerd default will be `runc` but that this should be used to specify alternative values like `kata-runtime` or `cc-runtime`.
The whole containerd runtime concept is complex enough and we should try to add the minimal amount of complexity with yet another runtime layer.

File: pkg/annotations/annotations.go [hunk 5]
```diff
@@ -31,4 +31,7 @@ const (
 
 	// SandboxID is the sandbox ID annotation
 	SandboxID = "io.kubernetes.cri.sandbox-id"
+
+	// PrivilegedSandboxis the privileged annotation
```
Comment (sameo) [inline pkg/annotations/annotations.go]: typo: s/PrivilegedSandboxis/PrivilegedSandbox is/

**Review by sameo — COMMENTED** [review]
A few minor comments, overall looks quite good.

**Pull request closed without merging**